packages/next-swc/crates/next-core/src/env.rs

```diff
@@ -61,6 +61,55 @@ pub async fn env_for_js(
         map.insert("__NEXT_STRICT_MODE_APP".to_string(), "true".to_string());
     }
 
+    map.insert(
+        "__NEXT_TRAILING_SLASH".to_string(),
+        if next_config.trailing_slash.unwrap_or(false) {
+            "true".to_string()
+        } else {
+            "false".to_string()
+        },
+    );
+
+    let base_path = next_config
+        .base_path
+        .clone()
+        .unwrap_or("undefined".to_string());
+    let defaulted_base_path = if base_path.is_empty() {
+        "undefined".to_string()
+    } else {
+        base_path
+    };
+
+    map.insert(
+        "__NEXT_ROUTER_BASEPATH".to_string(),
+        // Don't stringify undefined
+        if defaulted_base_path == "undefined" {
+            defaulted_base_path
+        } else {
+            serde_json::to_string(&defaulted_base_path).unwrap()
+        },
+    );
+
+    let asset_prefix = next_config
+        .asset_prefix
+        .clone()
+        .unwrap_or("undefined".to_string());
```

Comment:
That's weird. We don't need the sentinel value here and could use Option<String> instead